fuzz_test.go

```diff
@@ -504,3 +532,37 @@ func TestFuzz_SkipPattern(t *testing.T) {
 		return 5, true
 	})
 }
+
+type alwaysLast struct {
+}
+
+func (a alwaysLast) Int63n(n int64) int64 {
+	return n - 1
+}
+
+type alwaysFirst struct {
+}
+
+func (a alwaysFirst) Int63n(n int64) int64 {
```

Comment:
I don't super object to making the interface like you did, but go's embedding feature makes it pretty easy to produce a *rand.Rand with only a single method overridden like this:

```
type customInt63 struct {
  *rand.Rand

  mode int63mode
}

func (c customInt63) Int63n(n int64) int64 {
  switch c.mode {
  case modeFirst: return 0
  case modeLast: return n-1
  default: return c.Rand.Int63n(n)
  }
}
```

Reply:
I couldn't get the `type x struct { *rand.Rand }` code to work, hence why I introduced the interface. I'll try harder

Reply:
Oh wait-- rand.Rand isn't an interface. You're right, that's not going to work.

Reply:
Yeah let's stick with the interface you made and just split the seeding of the tests into another PR and we can merge this.

Reply:
Switched to use a custom object with a "mode" instead of two objects.